### ISSUE TITLE
type query arguments for querying within owned etc

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#14431]
### Summary of Problem
A feature request - to be able to use a query argument like `?t` to query the class type inside of an `owned` or possibly within other management types. E.g. `arg:owned ?t` or `arg: borrowed ?t`.

### Steps to Reproduce

``` chapel
proc f(arg:owned ?t) {
  writeln(t:string);
}
class MyClass { }
f(new owned MyClass());
```

Changing the formal to `arg:owned(?t)` produces different behavior.

**Associated Future Test(s):**
test/functions/ferguson/query/owned-inner1.chpl
test/functions/ferguson/query/owned-inner2.chpl

**Comment by bradcray** [comment]
I feel like we've had this conversation elsewhere, but I'm forgetting the answer:  Why does `owned(?t)` even parse?  (should it?)

**Comment by mppf** [comment]
`owned ?t` doesn`t parse today but `owned(?t)` does, and I don't have any answer for that other than "for historical reasons".  I am working on a PR that solves other type query problems (including #14314) but that disables both forms for the moment.

I don't think it makes sense to have `owned(?t)` only work - if we want `owned ?t` and `borrowed ?t` etc then we should build that. However I am thinking this is low priority (because it can already be done with `arg.type`).

**Comment by bradcray** [comment]
I agree `owned ?t` should be supported and was only asking about `owned(?t)` from the perspective of not wanting to get caught supporting it without good reason.  Disabling it sounds better than supporting it to me, thanks!

**Comment by e-kayrakli** [comment]
I think https://github.com/chapel-lang/chapel/pull/14668 can help with the parser side of the issue.

I kinda hoped that it would resolve completely, but it attempting to resolve it by simply adding few grammar rules didn't fix the issue. But made `owned ?t` and `owned(?t)` equivalent.
